lua/nvim-lsp-installer/ui/display.lua

```diff
@@ -99,42 +99,72 @@ end
 
 function M.new_view_only_win(name)
     local namespace = vim.api.nvim_create_namespace(("lsp_installer_%s"):format(name))
-    local buf, renderer, mutate_state, get_state, unsubscribe
+    local buf_handle, renderer, mutate_state, get_state, unsubscribe, win_id
     local has_initiated = false
 
     local function open(opts)
         opts = opts or {}
         local win_width, highlight_groups = opts.win_width, opts.highlight_groups
+        opts.layout = opts.layout or "popup"
 
-        if win_width then
-            vim.cmd(("%dvnew"):format(win_width))
+        if opts.layout == "popup" then
+            local win_height = vim.o.lines - vim.o.cmdheight - 2 -- Add margin for status and buffer line
+            win_width = vim.o.columns
+            local popup_layout = {
+                relative = "editor",
+                height = math.floor(win_height * 0.9),
+                width = math.floor(win_width * 0.8),
+                style = "minimal",
+                border = "rounded",
+            }
+            popup_layout.row = math.floor((win_height - popup_layout.height) / 2)
+            popup_layout.col = math.floor((win_width - popup_layout.width) / 2)
+            buf_handle = vim.api.nvim_create_buf(false, true)
```

Comment:
I feel like this should be the other way around, either

```suggestion
            buf_handle = vim.api.nvim_create_buf(true, false)
```

or 

```suggestion
            buf_handle = vim.api.nvim_create_buf(true, true)
```

There's no reason for the buffer to be listed, is there?

Reply:
Oh man, I do hate having these "magic" values sometimes, bu I've only seen it used like this so I never questioned it.

The way I have it now `vim.api.nvim_create_buf(false, true)` creates a *nobuflisted* and *scratch* buffer, isn't that correct?


```doc
nvim_create_buf({listed}, {scratch})

                Creates a new, empty, unnamed buffer.

                Parameters: ~
                    {listed}   Sets 'buflisted'
                    {scratch}  Creates a "throwaway" |scratch-buffer| for
                               temporary work (always 'nomodified'). Also sets
                               'nomodeline' on the buffer.
```

Reply:
Ahhh you're right.. got me booleans mixed up